modules/tracing/mule-tracer-configuration-impl/src/main/java/org/mule/runtime/tracing/level/impl/config/FileTracingLevelConfiguration.java

```diff
@@ -51,41 +52,53 @@
  */
 public class FileTracingLevelConfiguration extends FileConfiguration implements TracingLevelConfiguration, Disposable {
 
-  private final String CONFIGURATION_FILE_PATH =
-      getProperty(MULE_OPEN_TELEMETRY_TRACING_CONFIGURATION_FILE_PATH,
-                  getConfFolder() + FileSystems.getDefault().getSeparator() + getPropertiesFileName());
-
-  private final MuleContext muleContext;
-
   private static final String CONFIGURATION_FILE_NAME = "tracing-level.conf";
   private static final String LEVEL_PROPERTY_NAME = "mule.openTelemetry.tracer.level";
   private static final String OVERRIDES_PROPERTY_NAME = "mule.openTelemetry.tracer.levelOverrides";
+
   private static final Logger LOGGER = getLogger(FileTracingLevelConfiguration.class);
+  private final String CONFIGURATION_FILE_PATH =
+      getProperty(MULE_OPEN_TELEMETRY_TRACING_CONFIGURATION_FILE_PATH,
+                  getConfFolder() + FileSystems.getDefault().getSeparator() + getPropertiesFileName());
+
   private final HashMap<String, TracingLevel> tracingLevelOverrides = new HashMap<>();
-  private boolean tracingConfigurationFileWatcherInitialised;
+  private final List<Runnable> onConfigurationChangeRunnables = synchronizedList(new ArrayList<>());
+  private final LazyValue<TracingLevel> tracingLevelInitializer = new LazyValue<>(() -> {
+    initialise();
+    return null;
```

Comment:
why is this is a return null. make it return the corresponding tracinglevel.

Reply:
This LazyValue variable has the only task of running the `initialise()` method. There is no point in returning the tracing level because, as it will be executed only one time, the value could become outdated at some point. On the other hand, the tracing level is already managed in another variable and settled up by the `setTracingLevel()` method.